src/shared/Microsoft.Git.CredentialManager/PlatformUtils.cs

```diff
@@ -136,6 +137,24 @@ public static void EnsurePosix()
             }
         }
 
+        public static bool IsElevatedUser()
+        {
+            if (IsWindows())
+            {
+#if NETFRAMEWORK
+                var identity = System.Security.Principal.WindowsIdentity.GetCurrent();
+                var principal = new System.Security.Principal.WindowsPrincipal(identity);
+                return principal.IsInRole(System.Security.Principal.WindowsBuiltInRole.Administrator);
+#endif
```

Comment:
Just a question: is this available in the new, "unified" .NET 5? How much longer are we forced to use the .NET Framework?

Reply:
The API surface is in .NET Framwork and .NET 5 (and even in .NET Core 3.x and 1.x), but just not .NET _Standard_ 2.0. If we unified on .NET 5 then this `#if` could become just a runtime `if (PlatformUtils.IsWindows())`.

> How much longer are we forced to use the .NET Framework?

Sadly this looks to be much longer. There is one final problem with moving from .NET Framework to .NET 5 on Windows: File size and file count bloat.

Going from < 3MB and a dozen files.. to 100MB+ and 80+ files(!) which @dscho has shown some shock and horror at for bundling with Git for Windows (and rightly so!). Even with assembly trimming and linking, we're looking at maybe 40-50MB and then the last time I tried it, 1/2 of the UI features broke 😢

Reply:
Thanks for giving me some context. At minimum, this change doesn't add to the friction of a future switch.

Reply:
Yeah, in most cases the `#if`s are working around differences between what is and isn't in .NET Standard.
Moving to any concrete runtime is mostly just _deleting_ these conditions.